lib/active_model/serializer/adapter.rb

```diff
@@ -32,8 +34,38 @@ def self.adapter_class(adapter)
         "ActiveModel::Serializer::Adapter::#{adapter.to_s.classify}".safe_constantize
       end
 
+      def fragment_cache(*args)
+        raise NotImplementedError, 'This is an abstract method. Should be implemented at the concrete adapter.'
+      end
+
       private
 
+      def cache_check(serializer)
+        @serializer = serializer
```

Comment:
This mutation is breaking a test in https://github.com/ride/active_model_serializers/blob/master-rc1-with-fragment-cache/test/action_controller/json_api_linked_test.rb#L146-L150 (it's a branch which I'm backporting this into RC1 because that's what my company is using).

Basically, halfway through the serialization (within a execution of `serializable_hash`), `@serializer`, which was `PostSerializer`, becomes `AuthorSerializer` after this. That same test is passing in this PR, so I'm not really sure what's going on, but regardless of that, mutating a variable that's used to figure out how to serialize stuff is bad and can lead to nasty behaviors in the future. If we're going to mutate, we need to simply instantiate a new object then or make sure the serialization has finished.

@joaomdmoura if you're available, let's discuss this over chat :)